src/middleware/auth/auth.js

```diff
@@ -0,0 +1,29 @@
+const jwt = require('jsonwebtoken')
+const __ = require('../../util/response')
+const mongoose = require('mongoose')
+const userModel = require('../../models/Users')
+const { encryptorToken } = require('./encryptor')
+const { decryptorToken } = require('./decryptor')
+
+class Jwt {
+  async createToken (userId) {
+    userId = userId.toString()
+    return jwt.sign({ userId: encryptorToken(userId) }, process.env.SECRET_KEY, { expiresIn: '7d' })
+  };
+
+  async authentication (req, res, next) {
+    try {
+      const decoded = await jwt.verify(req.headers.authorization, process.env.SECRET_KEY)
+      if (decoded) {
+        const verifyUser = await userModel.__getUserById(mongoose.Types.ObjectId(decryptorToken(decoded.userId)))
+        if (!verifyUser) return res.status(401).json({ message: 'Illegal access' })
+        req.userId = verifyUser.userId
+        next()
+      };
+    } catch (error) {
+      return __.errorMsg(req, res, 401, error.message, error, 'authentication')
+    }
+  }
+};
+
+module.exports = new Jwt()
```

Comment:
Hey @whyDontI ,

Can you please explain to me how this ```js new Jwt()``` will be used in different controllers

Reply:
Hey @Swap76, we can simply use methods written in auth.js in other files by simply importing this file as ```const jwt = require('middleware/Auth/auth')```
Then, if I have to create JWT token I'll just write,
```const token = await jwt.createToken(user._id)```

Reply:
I'll use ```authentication()``` method to authenticate any request being received by the server.
I'll just have to pass it as an argument like below, 
```app.get('/', auth.authentication, UserService._getAllUsers) ```

Reply:
Cool good implementation of classes